src/foam/nanos/crunch/ServerCrunchService.java

```diff
@@ -205,6 +207,51 @@ public UserCapabilityJunction getJunction(X x, String capabilityId) {
     return this.getJunctionForSubject(x, capabilityId, subject);
   }
 
+  public boolean hasPreconditionsMet(
+    X x, String capabilityId
```

Comment:
Maybe a good practice would be to name 'x' something like either 'sysX' or 'sessionX', which would make reading code easier.